Frameworks/CoreLocation/CLLocation.mm

```diff
@@ -228,11 +224,33 @@ - (id)copyWithZone:(NSZone*)zone {
 }
 
 /**
- @Status Stub
+ @Status Interoperable
+ @Notes  This uses the "haversine" formula to find circular distance between two points, found
+         here: https://en.wikipedia.org/wiki/Haversine_formula
 */
 - (CLLocationDistance)distanceFromLocation:(const CLLocation*)location {
-    UNIMPLEMENTED();
-    return StubReturn();
+    // If either location is invalid return -1
+    if (location == nullptr) {
+        return -1.0;
+    }
+
+    if (!CLLocationCoordinate2DIsValid(self.coordinate) || !CLLocationCoordinate2DIsValid(location.coordinate)) {
+        return -1.0;
+    }
```

Comment:
should we log something here?  what does the reference platform on invalid args?